### PR TITLE
Validate all voucher properties

### PR DESCRIPTION
**Pull request by Hlavtox**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Validates all of the voucher properties to fix removal of invalid cart rules in front office. The PR is currently in draft because it has side effects we need to solve. autoRemoveFromCart is also called in backoffice from finished orders, and it removes all of the cart rules in every order change.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Indicate how to verify that this change works as expected.
| UI Tests          | Please run UI tests and paste here the link to the run. [Read this page to know why and how to use this tool.](https://devdocs.prestashop-project.org/8/contribute/contribution-guidelines/ui-tests/).
| Fixed issue or discussion?     | Fixes #{issue number here}, Fixes #{another issue number here}, Fixes https://github.com/PrestaShop/PrestaShop/discussions/ {discussion number here}
| Related PRs       | If theme, autoupgrade or other module change is needed to make this change work, provide a link to related PRs here.
| Sponsor company   | Your company or customer's name goes here (if applicable).

### CODE REVIEW

**Comment by Hlavtox** [comment]
@kpodemski This is basically it, we just have to use a different variable. `autoAddToCart` already supports two modes, the same goes for `checkValidity`, although the $useOrderPrices is a bit misleading.

Vouchers are now automatically removed on refresh and they work properly in backoffice.

File: classes/CartRule.php [hunk 1]
```diff
@@ -1669,7 +1669,7 @@ public function getAssociatedRestrictions(
      * Automatically add this CartRule to the Cart.
      *
      * @param Context|null $context Context instance
-     * @param bool $useOrderPrices
+     * @param bool $orderCreated
```
Comment (kpodemski) [inline classes/CartRule.php:1672]: Wrong docs.

**Review by kpodemski — COMMENTED** [review]
I'm not sure about it, `$alreadyInCart` and `$userOrderPrices`, those two have different meanings, and you've changed only the variables 🤔 I haven't tested it, and I see that tests are not run until you mark the PR as ready for a review.

I still think that having some extra abstraction there to better understand the context in which we operate would be cleaner.

**Comment by Hlavtox** [comment]
@kpodemski It's just a draw, but I think now we know what to do, what to validate where, and how to change it so we itroduce the least amount of BC breaks.

**Pull request closed without merging**